src/main/java/com/lmax/tool/disruptor/reflect/ReflectiveRingBufferInvocationHandler.java

```diff
@@ -53,7 +53,7 @@
     @Override
     public Object invoke(final Object proxy, final Method method, final Object[] args) throws Throwable
     {
-        if (TO_STRING_METHOD_NAME.equals(method.getName()))
+        if (TO_STRING_METHOD_NAME.equals(method.getName()) && args == null)
         {
```

Comment:
I think that you need to check args.length == 0.  Could you include a unit test to verify?